componentLibraries/defaultLibrary/Special/AeroComponents/AeroFlightRecorder.hpp

```diff
@@ -0,0 +1,329 @@
+        double roll;
+        double pitch;
+        double heading;
+        double speed;
+
+        //inputParameters
+        double dt;
+        double scaleX;
+        double scaleY;
+        double scaleZ;
+
+        // Constants
+        enum FileTypes {FlightPathKML, GroundPathKML, ThreeDmodelKML};
+        int mFileType;
+        HString mFilePath;
+        HString mDaeModelPath;
+        std::ofstream mFile;
+
+        // Other operating variables
+        double tilt;
+        double mLastLogTime;
+
+    public:
+    // The creator function that is registered when a component lib is loaded into Hopsan
+        static Component *Creator()
+        {
+            return new AeroFlightRecorder();
+        }
+
+        void configure()
+        {
+            //Add ports to the component
+
+            //Add inputVariables to the component
+            addInputVariable("longitude","Longitude E","deg",15.577,&mpLongitude);
+            addInputVariable("latitude","Latitude N","deg",58.401,&mpLatitude);
+            addInputVariable("altitude","Altitude","m",0,&mpAltitude);
+            addInputVariable("phi","Roll","rad",0,&mpPhi);
+            addInputVariable("theta","Pitch","rad",0,&mpTheta);
+            addInputVariable("psi","Heading","rad",0,&mpPsi);
+            addInputVariable("v","Speed","m/s",0,&mpV);
+
+            // Add inputParameters
+            addInputVariable("dt", "Logging interval (0 = log every sample)", "s", 1, &mpDt);
+            addInputVariable("scaleX", "X-axis scaling factor for .dae 3D model", "m", 1, &mpScaleX);
+            addInputVariable("scaleY", "Y-axis scaling factor for .dae 3D model", "m", 1, &mpScaleY);
+            addInputVariable("scaleZ", "Z-axis scaling factor for .dae 3D model", "m", 1, &mpScaleZ);
+
+            // Add conditional constant for choice of file type
+            std::vector<HString> filetypes;
+            filetypes.push_back("Flight path KML");
+            filetypes.push_back("Ground path KML");
+            filetypes.push_back("3D vehicle KML");
+            addConditionalConstant("filetype", "Output file type", filetypes,0,mFileType);
+
+            // Add constants
+            addConstant("path", "Path where the output file should be placed", "", "", mFilePath);
+            addConstant("daeModel", "File path to a .dae model for 3D vehicle representation", "", "", mDaeModelPath);
+        }
+
+
+        void initialize()
+        {
+            //Check if output file path has been specified
+            if (mFilePath.empty()) {
+                stopSimulation("Please specify a path for the output file.");
+                return;
+            }
+
+            //Fetch parent system name (hmf filename is unknown)
+            HString modelName = getSystemParent()->getName();
+
+            //Open file and write file headers depending on file type
+            if(mFileType == FlightPathKML) {
+                mFile.open((mFilePath.append("\\"+modelName+".kml")).c_str());
```

Comment:
This assumes Windows, unfortunately Hopsan does not have a filepath utltiy, as far as I remember.
But you could write something like this, and use sep+modelName

`#ifdef _WIN32`
`  const HString sep = "\\"`
`#else `
`  const HString sep = "/"`
`#endif`

Reply:
True! I didn't think about it. But that looks like a good solution. I'll add it!